### PR TITLE
C#: Update the make_stubs_nuget script

### PR DESCRIPTION
**Pull request by michaelnebel**
- Hardcode that we target the .NET 5.0 framework independently of what is the default on the machine, where the script is executed.
- Create a global.json file which hardcoded the use SDK to 5.0.402.
- Other minor changes.

### CODE REVIEW

File: csharp/ql/src/Stubs/make_stubs_nuget.py [hunk 1]
```diff
@@ -68,9 +68,13 @@ def write_csproj_prefix(ioWrapper):
     cmd.append(version)
 helpers.run_cmd(cmd)
 
+sdk_version = '5.0.402'
+print("\n* Creating new global.json file and setting SDK to " + sdk_version)
+helpers.run_cmd(['dotnet', 'new', 'globaljson', '--force', '--sdk-version', sdk_version])
```
Comment (hvitved) [inline csharp/ql/src/Stubs/make_stubs_nuget.py:73]: Shouldn't this command somehow mention `projectDirIn`?

File: csharp/ql/test/resources/stubs/_frameworks/Microsoft.NETCore.App/System.Runtime.cs [hunk 2]
```diff
@@ -5994,7 +5994,7 @@ public class Stopwatch
 
         namespace CodeAnalysis
         {
-            // Generated from `System.Diagnostics.CodeAnalysis.AllowNullAttribute` in `System.Diagnostics.DiagnosticSource, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51; System.Formats.Asn1, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51; System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a; System.Threading.Tasks.Dataflow, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a`
+            // Generated from `System.Diagnostics.CodeAnalysis.AllowNullAttribute` in `Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed; System.Diagnostics.DiagnosticSource, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51; System.Formats.Asn1, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51; System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a; System.Threading.Tasks.Dataflow, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a`
```
Comment (hvitved) [inline csharp/ql/test/resources/stubs/_frameworks/Microsoft.NETCore.App/System.Runtime.cs:5997]: This looks a bit weird, what do you think @tamasvajk ?

File: csharp/ql/src/Stubs/make_stubs_nuget.py (hunk 1 above)
Comment (michaelnebel) [reply csharp/ql/src/Stubs/make_stubs_nuget.py]: dotnet looks traverses the directory structure upwards towards the root from the current work dir to find a global.json file.
We need to change the working directory to the tempDir and store the global.json here as we execute dotnet commands against multiple subfolders in the the tempDir.

Comment (michaelnebel) [reply csharp/ql/src/Stubs/make_stubs_nuget.py]: @hvitved : But you are right - without any changes we will start poluting the folder from which the nuget script is executed.

File: csharp/ql/test/resources/stubs/_frameworks/Microsoft.NETCore.App/System.Runtime.cs (hunk 2 above)
Comment (tamasvajk) [reply csharp/ql/test/resources/stubs/_frameworks/Microsoft.NETCore.App/System.Runtime.cs:5997]: Yes, I think we would manually need to touch up the files in `_frameworks` folder, and remove any mention of `Newtonsoft.Json`.

What happens here is that `System.Diagnostics.CodeAnalysis.AllowNullAttribute` is defined in multiple assemblies, and one of them happens to be `Newtonsoft.Json`, the others are all in the runtime.

File: csharp/ql/src/Stubs/make_stubs_nuget.py [hunk 3]
```diff
@@ -106,7 +109,7 @@ def write_csproj_prefix(ioWrapper):
 print("\n --> Generated stub files: " + rawSrcOutputDir)
 
 print("\n* Formatting files")
-helpers.run_cmd(['dotnet', 'format', rawSrcOutputDir])
+run_cmd(['dotnet', 'format', rawSrcOutputDir], workDir)
```
Comment (hvitved) [inline csharp/ql/src/Stubs/make_stubs_nuget.py:112]: spurious `workDir` arg?

File: csharp/ql/test/resources/stubs/_frameworks/Microsoft.NETCore.App/System.Runtime.cs (hunk 2 above)
Comment (michaelnebel) [reply csharp/ql/test/resources/stubs/_frameworks/Microsoft.NETCore.App/System.Runtime.cs:5997]: Alright. I will just roll-back the changes made in this file as the only difference between the previous version and this one is the mentioning of Newtonsoft.Json.

File: csharp/ql/src/Stubs/make_stubs_nuget.py (hunk 3 above)
Comment (michaelnebel) [reply csharp/ql/src/Stubs/make_stubs_nuget.py:112]: Very nice catch @hvitved! Now the directory would be printed as error message... Will fix!

**Review by hvitved — APPROVED** [review]

**Pull request merged**